Mathlib/Topology/UniformSpace/Completion.lean

```diff
@@ -383,9 +383,14 @@ instance : SeparatedSpace (Completion α) :=
 instance : T3Space (Completion α) :=
   separated_t3
 
+/-- The map from a uniform space to its completion.
+
+porting note: this was added to create a target for the `@[coe]` attribute. -/
+@[coe] def coe' : α → Completion α := Quotient.mk' ∘ pureCauchy
+
 /-- Automatic coercion from `α` to its completion. Not always injective. -/
-instance : CoeTC α (Completion α) :=
-  ⟨Quotient.mk' ∘ pureCauchy⟩
+instance : Coe α (Completion α) :=
```

Comment:
Have I done the right thing here?

Before this change, the coercion was being eagerly unfolded and causing the usual problems (e.g., with rewrites).

I note also that the guidance [from core](https://github.com/leanprover/lean4/blob/3b50410ec0a8b83344d0f61df554a1024bdaf298/src/Init/Coe.lean#L133) regarding `CoeTC` is that:
```quote
Auxiliary class implementing `Coe*`.
Users should generally not implement this directly.
```

Reply:
My imporession is that adding the `@[coe] def coe'` above is sufficient, and that you don't need this instance at all. But I could be very wrong about that.

Reply:
I wasn't clear on this myself but your comment motivated me to look this up!

Based on what I see [here](https://github.com/leanprover/std4/blob/ac39f359c64f54e833e358cf0a6b3ca9f00aa490/Std/Tactic/CoeExt.lean#L89-L103) I now understand that both the `@[coe]` and the `instance` are required.